pkg/zack/layout_split.go

```diff
@@ -0,0 +1,254 @@
+			// if the start view has no width, return the end view only (fullscreen-ish)
+			return endContent
+		}
+		if m.endChildBounds.Width <= 0 {
+			// if the end view has no width, return the start view only (fullscreen-ish)
+			return startContent
+		}
+		return lipgloss.JoinHorizontal(lipgloss.Top,
+			renderWithBounds(&m.startChildBounds, startContent),
+			renderWithBounds(&m.endChildBounds, endContent))
+
+	case Vertical:
+		if m.startChildBounds.Height <= 0 {
+			return endContent
+		}
+		if m.endChildBounds.Height <= 0 {
+			return startContent
+		}
+		return lipgloss.JoinVertical(lipgloss.Left,
+			renderWithBounds(&m.startChildBounds, startContent),
+			renderWithBounds(&m.endChildBounds, endContent))
+	}
+
+	log.Warn().Msgf("(SplitLayout.Render) Invalid SplitLayoutModel orientation: %v", m.orientation)
+	return ""
+}
+
+// SetBounds sets the bounds for the start and end child views based on the parent bounds.
+func (m *SplitLayoutModel) SetBounds(bounds Bounds) {
+	m.layoutBounds = bounds
+
+	switch m.orientation {
+	case Horizontal:
+		var startWidth, endWidth int
+		if m.fixedStartSize > 0 {
+			if m.fixedEndSize > 0 {
+				log.Warn().Msgf("(%s) Both fixedStartSize and fixedEndSize are set, using fixedStartSize: %d",
+					"SplitLayoutModel.SetBounds",
+					m.fixedStartSize)
+			}
+			startWidth = int(m.fixedStartSize)
+			endWidth = bounds.Width - startWidth
+		} else if m.fixedEndSize > 0 {
+			endWidth = int(m.fixedEndSize)
+			startWidth = bounds.Width - endWidth
+		} else {
+			startWidth = int(float64(bounds.Width) * m.fraction)
+			endWidth = bounds.Width - startWidth
+		}
+		m.setStartChildBounds(NewBounds(startWidth, bounds.Height))
+		m.setEndChildBounds(NewBounds(endWidth, bounds.Height))
+		return
+
+	case Vertical:
+		var startHeight, endHeight int
+		if m.fixedStartSize > 0 {
+			startHeight = int(m.fixedStartSize)
+			endHeight = bounds.Height - startHeight
+		} else if m.fixedEndSize > 0 {
+			endHeight = int(m.fixedEndSize)
+			startHeight = bounds.Height - endHeight
+		} else {
+			startHeight = int(float64(bounds.Height) * m.fraction)
+			endHeight = bounds.Height - startHeight
+		}
+		m.setStartChildBounds(NewBounds(bounds.Width, startHeight))
+		m.setEndChildBounds(NewBounds(bounds.Width, endHeight))
+		return
+	}
+
+	log.Warn().Msgf("(SplitLayout.SetBounds) Invalid SplitLayoutModel orientation: %v", m.orientation)
+}
+
+// refreshBounds send the newest bounds to the start and end child boundables.
+// This is useful when the _layout_ itself changes, e.g. when you increase or decrease the fraction,
+func (m *SplitLayoutModel) refreshBounds() {
+	m.SetBounds(m.layoutBounds)
+}
+
+// setStartChildBounds sets the bounds for the start child view and updates its boundable.
+func (m *SplitLayoutModel) setStartChildBounds(bounds Bounds) {
+	m.startChildBounds = bounds
+	if !bounds.IsValid() {
+		log.Warn().Msgf("(SplitLayout.setStartChildBounds) start child bounds are invalid: %v", bounds)
+		return
+	}
```

Comment:
Should we not move the assignment of the bounds below the `IsValid()` check?
```suggestion
	if !bounds.IsValid() {
		log.Warn().Msgf("(SplitLayout.setStartChildBounds) start child bounds are invalid: %v", bounds)
		return
	}
	m.startChildBounds = bounds

```